pkg/cluster/update.go

```diff
@@ -104,9 +107,24 @@ func (u *simpleUpgrader) waitForNewNodes(ctx context.Context, cs *api.OpenShiftM
 				if err != nil {
 					return err
 				}
+				blob[instanceName(*vm.Name)] = ssHashes[scalesetName(agent.Name)]
+				if err := u.updateBlob(blob); err != nil {
+					return err
+				}
+			}
+			// this VM has been removed from the scale set but still around
+			// in the blob hence it's a stale one we should cleanup.
+			if _, ok := blob[instanceName(*vm.Name)]; !ok {
+				staleVMs = append(staleVMs, instanceName(*vm.Name))
 			}
 		}
 	}
+	if len(staleVMs) > 0 {
```

Comment:
... and now, iterate *blobs*.  If you find something not in the slice, delete it.